src/towncrier/check.py

```diff
@@ -101,12 +101,14 @@ def __main(
         click.echo(f"{n}. {change}")
     click.echo("----")
 
+    # This will fail if any fragment files have an invalid name:
```

Comment:
I don't think that this comment is needed.

The docstring for `find_fragments` should be enough of a documentation.

Reply:
I added this comment to make sure anyone thinks twice before possibly moving this back down below to just before the result is used

Reply:
I think this will fail because of the control case fragment, but I haven't tried moving the line down and re-running

https://github.com/twisted/towncrier/blob/3c93f134eda738ca7e27eef541b08448bec80d88/src/towncrier/test/test_check.py#L493-L503

Reply:
We need to check this, as the test are the critical part of a PR.

Reply:
Done (test updated to ensure)

Reply:
Thanks. I didn't had time to check the test coverage... but if say that you have checked it, we can have this merged :)

no problem